python/lsst/ts/externalscripts/auxtel/correct_pointing.py

```diff
@@ -0,0 +1,290 @@
+
+    @classmethod
+    def get_schema(cls):
+        return None
+
+    def set_metadata(self, metadata):
+        pass
+
+    async def configure(self, config):
+        """Configure script.
+
+        Parameters
+        ----------
+        config : `types.SimpleNamespace`
+            Script configuration, as defined by `schema`.
+        """
+        self.best_effort_isr = self.get_best_effort_isr()
+
+    def get_best_effort_isr(self):
+        # Isolate the BestEffortIsr class so it can be mocked
+        # in unit tests
+        return BestEffortIsr()
+
+    @property
+    def camera_readout_time(self):
+        return 2.0
+
+    async def arun(self, checkpoint_active=False):
+        """Execute script main operations.
+
+        Parameters
+        ----------
+        checkpoint_active : bool, optional
+            Run script with checkpoints, by default False?
+        """
+
+        await self.handle_checkpoint(
+            checkpoint_active=checkpoint_active,
+            checkpoint_message="Setting up.",
+        )
+
+        await asyncio.gather(
+            self._setup_atcs(),
+            self._setup_latiss(),
+        )
+
+        await self.handle_checkpoint(checkpoint_active, "Correcting pointing...")
+
+        await self.execute_grid(90.0, 60.0, 0.0)
+
+        offset_summary = await self.atcs.rem.atptg.evt_offsetSummary.aget(
+            timeout=self.atcs.long_timeout
+        )
+
+        porigin_x = (
+            offset_summary.pointingOriginX + offset_summary.pointingOriginHandsetDX
+        )
+        porigin_y = (
+            offset_summary.pointingOriginY + offset_summary.pointingOriginHandsetDY
+        )
+
+        self.log.info(f"{porigin_x=}, {porigin_y=}")
+
+        await self.atcs.reset_offsets()
+
+        await self.atcs.rem.atptg.cmd_poriginXY.set_start(
+            x=porigin_x, y=porigin_y, timeout=self.atcs.long_timeout
+        )
+
+    async def _setup_latiss(self):
+        """Setup latiss.
+
+        Set filter and grating to empty_1.
+        """
+        await self.latiss.setup_instrument(
+            filter="empty_1",
+            grating="empty_1",
+        )
+
+    async def _setup_atcs(self):
+        """Setup ATCS.
+
+        This method will reset the ATAOS and pointing offsets.
+        """
+        self.log.info("Resetting pointing and hexapod x and y offsets.")
+
+        await self.atcs.rem.ataos.cmd_resetOffset.set_start(
+            axis="x",
+            timeout=self.atcs.long_timeout,
+        )
+        await self.atcs.rem.ataos.cmd_resetOffset.set_start(
+            axis="y",
+            timeout=self.atcs.long_timeout,
+        )
+        await self.atcs.reset_offsets()
+
+    async def handle_checkpoint(self, checkpoint_active, checkpoint_message):
+
+        if checkpoint_active:
+            await self.checkpoint(checkpoint_message)
+        else:
+            self.log.info(checkpoint_message)
+
+    async def execute_grid(self, azimuth, elevation, rotator):
```

Comment:
why "execute_grid" -- presumably this comes from your pointing model determination script..?